### PR TITLE
feat: In-app rating on Android (Google Play)

### PR DESCRIPTION
**Pull request by morning4coffe-dev**
GitHub Issue (If applicable): 

## PR Type

What kind of change does this PR introduce?

- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	<ItemGroup>
+		<PackageReference Include="Xamarin.Google.Android.Play.Core" Version="1.10.3.14" />
+
+		<ProjectReference Include="..\..\Uno.Foundation.Logging\Uno.Foundation.Logging.csproj" />
+		<ProjectReference Include="..\..\Uno.UI\Uno.UI.netcoremobile.csproj" />
+	</ItemGroup>
+
+	<Import Project="..\..\SourceGenerators\Uno.UI.SourceGenerators\Content\Uno.UI.SourceGenerators.props" />
+
+	<Target Name="_CleanupReferences" BeforeTargets="CoreCompile">
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj:39]: Can this target be removed for this project?

File: src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			<Analyzer Remove="@(Analyzer)" Condition="'%(Analyzer.NugetPackageId)'=='Uno.WinUI'" />
+		</ItemGroup>
+	</Target>
+
+	<Target Name="_UnoToolkitOverrideNuget" AfterTargets="AfterBuild" DependsOnTargets="BuiltProjectOutputGroup" Condition="'$(UnoNugetOverrideVersion)'!=''">
+
+		<PropertyGroup>
+			<_OverrideTargetFramework>$(TargetFramework)</_OverrideTargetFramework>
+			
+			<!-- Current nuget publishing uses netX.0-android30.0 -->
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj:52]: This package is likely going to be without platform version using MSBuild packaging

File: src/AddIns/Uno.UI.GooglePlay/buildTransitive/Uno.WinUI.GooglePlay.targets [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+	<ItemGroup>
+		<UnoRuntimeEnabledPackage Include="Uno.UI.GooglePlay"  PackageBasePath="$(MSBuildThisFileDirectory)" Condition="'$(MSBuildThisFile)'=='uno.ui.googleplay.targets'" />
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/buildTransitive/Uno.WinUI.GooglePlay.targets:5]: I don't think this should exist for this package

File: src/AddIns/Uno.UI.GooglePlay/StoreContextExtension.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().LogError("Error", "There was an error launching in-app review. Please try again.");
+				}
+
+				_inAppRateTcs?.TrySetResult(new(StoreRateAndReviewStatus.Error));
+
+				System.Diagnostics.Debug.WriteLine($"Error message: {ex.Message}");
+				System.Diagnostics.Debug.WriteLine($"Stacktrace: {ex}");
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/StoreContextExtension.cs:94]: Should this info be part of the LogError call above?

File: src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	<ItemGroup>
+		<PackageReference Include="Xamarin.Google.Android.Play.Core" Version="1.10.3.14" />
+
+		<ProjectReference Include="..\..\Uno.Foundation.Logging\Uno.Foundation.Logging.csproj" />
+		<ProjectReference Include="..\..\Uno.UI\Uno.UI.netcoremobile.csproj" />
+	</ItemGroup>
+
+	<Import Project="..\..\SourceGenerators\Uno.UI.SourceGenerators\Content\Uno.UI.SourceGenerators.props" />
+
+	<Target Name="_UnoToolkitOverrideNuget" AfterTargets="AfterBuild" DependsOnTargets="BuiltProjectOutputGroup" Condition="'$(UnoNugetOverrideVersion)'!=''">
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj:34]:
```suggestion
	<Target Name="_UnoGooglePlayOverrideNuget" AfterTargets="AfterBuild" DependsOnTargets="BuiltProjectOutputGroup" Condition="'$(UnoNugetOverrideVersion)'!=''">
```

File: src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+		<PropertyGroup>
+			<_TargetNugetFolder>$(NuGetPackageRoot)\uno.ui.googleplay\$(UnoNugetOverrideVersion)\lib\$(TargetFramework)</_TargetNugetFolder>
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj:37]:
```suggestion
			<_TargetNugetFolder>$(NuGetPackageRoot)\uno.winui.googleplay\$(UnoNugetOverrideVersion)\lib\$(TargetFramework)</_TargetNugetFolder>
```

File: src/Directory.Build.props [hunk 7]
```diff
@@ -123,6 +123,8 @@
 		<_AdjustedOutputProjects>$(_AdjustedOutputProjects);Uno.UI.Toolkit.Reference.csproj</_AdjustedOutputProjects>
 		<_AdjustedOutputProjects>$(_AdjustedOutputProjects);Uno.UI.Toolkit.Windows.csproj</_AdjustedOutputProjects>
 
+		<_AdjustedOutputProjects>$(_AdjustedOutputProjects);Uno.UI.GooglePlay.netcoremobile.csproj</_AdjustedOutputProjects>
```
Comment (Youssef1313) [inline src/Directory.Build.props:126]: This shouldn't be needed

File: src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	</PropertyGroup>
+
+	<Import Project="../../targetframework-override.props" />
+
+	<PropertyGroup>
+		<AssemblyName>Uno.UI.GooglePlay</AssemblyName>
+		<RootNamespace>Uno.UI.GooglePlay</RootNamespace>
+		<PackageId Condition="'$(UNO_UWP_BUILD)'!='true'">Uno.WinUI.GooglePlay</PackageId>
+
+		<GeneratePackageOnBuild>true</GeneratePackageOnBuild>
```
Comment (Youssef1313) [inline src/AddIns/Uno.UI.GooglePlay/Uno.UI.GooglePlay.netcoremobile.csproj:14]: Only in Release please

**Review by MartinZikmund — APPROVED** [review]

File: build/Uno.UI.Build.csproj [hunk 9]
```diff
@@ -368,6 +368,8 @@
 		<ItemGroup>
 			<_diffPackage Include="$(PackageNamePrefix)" Other="$(PackageNamePrefix).$(GITVERSION_SemVer).nupkg" />
 			<_diffPackage Include="$(PackageNamePrefix).Foldable" Other="..\..\..\NugetPackages-Artifacts-netcoremobile-$(ArtifactsPlatformName)\vslatest-netcoremobile\$(PackageNamePrefix).Foldable.$(GITVERSION_SemVer).nupkg" />
+			<!-- TODO: Skip Package diff before released on Nuget -->
+			<!-- <_diffPackage Include="$(PackageNamePrefix).GooglePlay" Other="..\..\..\NugetPackages-Artifacts-netcoremobile-$(ArtifactsPlatformName)\vslatest-netcoremobile\$(PackageNamePrefix).GooglePlay.$(GITVERSION_SemVer).nupkg" /> -->
```
Comment (MartinZikmund) [inline build/Uno.UI.Build.csproj:372]: After merged, make PR to re-add this @morning4coffe-dev

**Comment by MartinZikmund** [comment]
@morning4coffe-dev Please also add documentation on how to add this package to Uno app to actually be able to use the API on Android

**Pull request merged**